frontend/src/pages/PipelineDetails.tsx

```diff
@@ -666,8 +666,13 @@ class PipelineDetails extends Page<{}, PipelineDetailsState> {
         const template = JsYaml.safeLoad(templateString);
         if (WorkflowUtils.isArgoWorkflowTemplate(template)) {
           graph = StaticGraphParser.createGraph(template!);
```

Comment:
If I understand it correctly, you would probably like to implement the validation for `graph` here instead of `reducedGraph`, because it is the default rendering approach. and the `transitiveReduction()` is like a post-render action to check whether the graph has been too large.